### ISSUE TITLE
ci(codeql): code-scanning digest — 3 open alerts

### ISSUE THREAD

**Issue by script-kid-io** [zeroroot-ai/gibson#1436]
## Code-scanning digest — 2026-08-15

Open alerts in `zeroroot-ai/gibson`: **3** total — 0 error, 3 warning, 0 note/other.

No error-severity alerts are open.

Warning/note alerts roll up into the counts above and are not listed individually.

Full list: https://github.com/zeroroot-ai/gibson/security/code-scanning?query=is%3Aopen

_Auto-filed on a schedule by the org SARIF-triage workflow (zeroroot-ai/.github). One digest issue per repo, found by title and updated in place; repos with zero open alerts are skipped. Fix alerts to close them, or dismiss them with a reason per slice 4.6 suppression discipline (ADR-0013)._

**Comment by script-kid-io** [comment]
## Verification 2026-08-15 — NOT stale, and the count is a 65x undercount

Triage swept this as a possible stale digest. It is the opposite: the title says 3 open alerts, the live count is **195**.

```
gh api --paginate repos/zeroroot-ai/gibson/code-scanning/alerts?state=open --jq '.[].number' | wc -l
195
```

By tool: **CodeQL 0, Trivy 195.** By severity: 4 critical / 21 high / 67 medium / 67 low.

Two things this means:

1. **The digest is scoped to the wrong tool.** It is titled `ci(codeql)` and reports a CodeQL-shaped count, while every open alert on this repo now comes from Trivy. A digest that reads 3 while 195 alerts are open is worse than no digest, because it reads as reassurance.
2. **The spike is expected, not a regression.** Code-scanning counts across zeroroot-ai were frozen for roughly three months by scanning defects. Those were fixed, and real counts are surfacing now. A jump from 3 to 195 is the freeze lifting, not 192 new vulnerabilities landing today.

Details deliberately omitted here — this is a public repo. Full list: https://github.com/zeroroot-ai/gibson/security/code-scanning?query=is:open

Remaining work is (a) fix the digest generator's tool scope in `zeroroot-ai/.github` (it was rewritten in .github#253; the tool-scope defect is separate), and (b) triage the 25 critical+high alerts. Keeping this open and relabeling accordingly.

**Comment by script-kid-io** [comment]
## Triage 2026-08-15 — `post-launch` (label only; the issue stays open)

Following the verification comment above, applying the rubric's runtime-code-vs-hygiene split: **first-party runtime code (`app/`, `internal/`, `cmd/`) is fix-only and launch-blocking; base-image and dependency findings are post-launch.**

Where the 195 actually are, by location:

| location | alerts |
|---|---|
| `zeroroot-ai/gibson-belief-sidecar` image | **180** — including all 4 criticals and 19 of the 21 highs |
| `usr/local/lib/.../pip-25.0.1.dist-info/METADATA` | 5 |
| gibson's own binaries (`gibson`, `gibson-migrate`, `ext-authz`, `manager`, `backfill-*`) | 10, of which 1 high and 2 medium |

CodeQL — the tool that scans gibson-authored Go — has **0** open alerts. The pile is a Python/Debian base-image dependency graph in one sidecar image, not findings in code this repo wrote. That is the post-launch bucket.

Two consequences worth acting on when this is picked up:

1. The single highest-leverage fix is rebasing / slimming the belief-sidecar image, not per-alert triage — 92% of the count and 100% of the criticals are in that one image.
2. The generator defect stands: the digest is titled and scoped `ci(codeql)` while every open alert comes from Trivy. Fix belongs in `zeroroot-ai/.github`.

**Comment by script-kid-io** [comment]
## Worked this digest — 24 of the 25 HIGH/CRITICAL are one image, and none of them are our code

Re-pulled the alert list from the code-scanning API today rather than reading it off this issue. The 195 total is accurate. The severity picture is much more concentrated than the table above suggests:

| Where | crit | high | What it is |
|---|---|---|---|
| `gibson-belief-sidecar` | 4 | 20 | Debian userland — `perl-base`, `util-linux` + its libs, `ncurses`, `gzip`, `libacl1`, plus `pip` |
| `usr/local/bin/gibson` | 0 | 1 | `github.com/buger/jsonparser v1.1.1`, CVE-2026-32285 |

That is the whole HIGH/CRITICAL set. **Nothing in it is first-party code.**

### The sidecar: two causes, both removed — #1464

**The Debian userland.** `python:3.12-slim` ships 96 dpkg packages. Four CRITICALs in `perl-base` alone, and nothing in the image invokes perl. Critically, **most of these carry no fixed version** — the `Fixed Version:` field is empty on 23 of them — so "bump the base image" was never an available fix. Not shipping a Debian userland is the only remedy.

Runtime stage is now `distroless/python3-debian13:nonroot`: 37 packages, no perl, no util-linux tooling, no gzip, no shell, no package manager, no pip. Verified by exporting both images and diffing `usr/share/doc/*`.

**pgmpy pulls torch.** Worth stating plainly because it is not visible from the alert list: `requirements.txt` was three lines and resolved to **44 packages** — `torch`, `triton`, `xgboost`, `scikit-learn`, `pandas`, `scipy`, `statsmodels`, `sympy`. Roughly 3 GB of a deep-learning stack, in a production image, to marginalise a **seven-node binary network**. The service does no training, no autodiff and no tensor work.

Inference now runs on `infer.py` — the same algorithm (sum-product variable elimination) on numpy alone. Variable elimination *is* exact inference, so this is not an approximation: the elimination order changes the cost, never the answer. `test_parity.py` asserts agreement with `pgmpy==0.1.26` to **1e-12** over the shipped artifact's entire evidence space and 25 random DAGs, and **that test passed in CI on this PR** — it is evidence, not a claim.

### The one real dependency finding

`github.com/buger/jsonparser v1.1.1` → CVE-2026-32285, fixed in 1.1.2, currently an indirect requirement. Being handled separately.

### Two things found while doing this

1. **`belief-sidecar-test` never ran on a pull request.** `gibson-images.yml` triggers only on `push: main`, tags and dispatch, and `sidecar/belief` is Python so no go-ci lane compiles it. A change to that directory reached `main` with no gate having run on the PR at all. Fixed in #1464 by extracting the job into `belief-sidecar.yml`, consumed on `pull_request` **and** via `workflow_call` as the publish gate.

2. **The mirror never refreshed the sidecar's base.** The `python` entry in `zeroroot-ai/.github`'s `mirror-list.yaml` had no `refresh: true`, so the default skip-if-exists behaviour froze the tag at whatever patch it was first copied at. Debian security updates never reached the sidecar even in the cases where a fix existed. Fixed in zeroroot-ai/.github#273.

### Left for `enterprise/deploy` — deliberately not touched

The chart's `startupProbe` for this container allows 5 minutes (`failureThreshold: 60`) for "loading the pgmpy model". With a numpy-only load that is wildly over-provisioned and should come down. `enterprise/deploy` is frozen while chain 1 is in flight, so it is not in #1464.

**Comment by script-kid-io** [comment]
## Measured after #1464 merged. Reporting what the scan actually says, not what I expected it to say.

`#1464` is on `main` (`d605a641`); the image built and published from it, and the parity gate passed. Re-pulled the alerts from the code-scanning API against that build:

| | before | after |
|---|---|---|
| **critical** | **4** | **0** |
| high | 21 | 16 |

**All four CRITICALs are gone** — they were all `perl-base`. So are `util-linux`'s tooling packages, `gzip`, `libacl1` and `pip`.

### The high count did not fall as far as "24 alerts cleared" would suggest, and the reason matters

Twelve of the sixteen remaining are **CPython itself** — `python3.13-minimal`, `libpython3.13-minimal`, `libpython3.13-stdlib`, `python3.13-venv`, three CVEs each. Those are not new problems. They are newly **visible** ones.

`python:3.12-slim` builds CPython from source into `/usr/local`, so it is not a dpkg package and Trivy could not attribute a CVE to it — the interpreter's vulnerabilities were present and unscannable. (That is also why the old alert list showed `pip` at `usr/local/lib/python3.12/site-packages/...` but no interpreter entry at all.) In distroless, CPython *is* a dpkg package, so its CVEs are now reported.

That is a better posture, not a worse one — but it means the honest headline is "4 critical and the entire Debian userland removed, and the interpreter's own CVEs became visible", not "24 alerts cleared".

### Every remaining sidecar finding has no fix

| CVE | Package | Fixed version |
|---|---|---|
| CVE-2026-11940, CVE-2026-15308, CVE-2026-7210 | `python3.13-*` / `libpython3.13-*` (3.13.5-2+deb13u4) | **none** |
| CVE-2025-69720 | `libncursesw6`, `libtinfo6` | **none** |
| CVE-2026-53615 | `libuuid1` | **none** |

There is nothing to bump to for any of them. `libncursesw6` / `libtinfo6` / `libuuid1` are the three I expected to survive — they back CPython's optional `readline`, `_curses` and `_uuid` extension modules, none of which this sidecar imports. Removing them would mean building CPython ourselves with those modules disabled; that is a real option and a bigger change, and it would not touch the twelve interpreter CVEs anyway.

Because the base is `refresh: true` in the mirror now (zeroroot-ai/.github#273), a Debian fix for any of these reaches the sidecar on the next mirror run instead of being frozen forever. That was not true before.

### The last alert outside the sidecar

`github.com/buger/jsonparser v1.1.1` → CVE-2026-32285, fixed in 1.1.2. #1465 takes it to v1.6.1.

### A gate that can never go green

`gibson-belief-sidecar / Vulnerability scan` failed on this run. It is not a regression from #1464 — **it failed identically on the previous run** (`31958670203`, before the merge), and it fails by construction: `trivy-action` is configured `severity: HIGH,CRITICAL` with `exit-code: 1`, so any unfixable third-party HIGH keeps it red permanently.

That is the mirror image of the guards-that-cannot-fail class this repo keeps finding: a gate that cannot pass carries no signal either, and a real regression landing next to it looks exactly like the known-red baseline. Not fixed here — it needs a decision about whether unfixable findings should be dismissed with a reason (per the ADR-0013 suppression discipline) or the gate scoped to fixable findings only.
